### ISSUE TITLE
APNS delegate never fired

### ISSUE THREAD

**Issue by atilsamancioglu** [OneSignal/OneSignal-iOS-SDK#514]
**Description:**

APNS delegate never fired in Xcode 11 beta 5

**Environment**

Xcode 11 Beta5

This is my pod file:

target 'MyApp' do
  use_frameworks!

pod 'Firebase/Analytics'
pod 'Firebase/Auth'
pod 'Firebase/Core'
pod 'Firebase/Firestore'
pod 'Firebase/Storage'
pod 'SDWebImage'
pod 'OneSignal', '>= 2.6.2', '< 3.0'

target 'OneSignalNotificationServiceExtension' do
  pod 'OneSignal', '>= 2.6.2', '< 3.0'
end

end

**Steps to Reproduce Issue:**

1. pod install
1. pod update
1. try to register an iPhone device

**Anything else:**

I've been trying to make this work in Xcode 11 beta5. first we had [this](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/2.10.1) issue now it's resolved. I updated the pod and it seemed OK for a while. I managed to subscribe my phone but after a while, out of blue, it shows me APNS delegate never fired  in all users and my phone is unsubscribed anyhow.

**Comment by Nightsd01** [comment]
@atilsamancioglu can you describe your issue a bit more especially the APNS delegate..? Is the issue basically that your phone is now unsubscribed and you’re not getting push notifications?

**Comment by atilsamancioglu** [comment]
hi, yes it gets unsubscribed but I don't get any error in the logs, it just happens out of nowhere. I only see this "APNS delegate never fired" message when I go to all users tab in dashboard.

**Comment by jkasten2** [comment]
I did some testing on iPadOS 13 Beta 7 with Xcode 11 Beta 5 and it seems for some reason when calling ` [[UIApplication sharedApplication] registerForRemoteNotifications];` APNs never responses.
Neither of the AppDelegate methods are never called by iOS sometimes to indicate a success OR failure;
- `application:didFailToRegisterForRemoteNotificationsWithError`
- `application:didRegisterForRemoteNotificationsWithDeviceToken`

Even restarting the app or uninstalling it and reinstalling does not help. However what seems to get things working is toggling WiFi. This seems to indicate something changed in iOS on how it is handling connects to Apple's APNs server.

We are continuing to test and will provide more details here.

**Comment by atilsamancioglu** [comment]
yes exactly, toggling wifi seemed to do the trick for me too

**Comment by jkasten2** [comment]
I have been testing on two different iPadOS devices and have yet to see this issue again. Not even a slow response which I was seeing before sometimes.

I suspect this was some kind of issue on Apple's servers, possibly they have different servers handling responses for the iOS 13 beta.

@atilsamancioglu Let us know if you are able to reproduce the issue again. If so I recommend getting the full system log from Window > "Devices and Simulators" and press the "Open Console" for your device. The log should then be shared with Apple.

**Comment by markosole** [comment]
Hi guys, I am having similar problem where Wifi reset did not help. When App asks me do I allow to receive notifications, I confirm with Allow and second after I get this on the log:

`Getting notification settings (async)
Got notification settings [ hasResult: 1 hasCompletionHandler: 1 ]
Getting badge number (sync)
Got badge number [ badge: (null) ]`

On the OneSignal dashboard I can see all details about SDK version, build version, OS version etc. But phone is not getting `PUSH TOKEN`.

Xcode: 11
iOS device: 13.1.3

**Here is what I have in the app.js**

        if (application.ios) {
            let opts = NSMutableDictionary.new();
            TnsOneSignal.initWithLaunchOptionsAppIdHandleNotificationReceivedHandleNotificationActionSettings({},
                    'MY_TOKEN',
                    (notificationReceived) => {
                        console.log(notification);
                    },
                    (notificationAction) => {
                        console.log(notification);
                    },
                    // (settings) => {
                    //    // console.log(notification);
                    // }
                    opts
                    );
        }

I had to add `opts` because app was crashing and saying that 5th parameter is missing. I don't know if this is ok or I need to use something else instead.

Here is what I have installed:

![image](https://user-images.githubusercontent.com/10780405/68094115-3b4d1380-fe95-11e9-8e2b-d64cc93630c7.png)

**Comment by jkasten2** [comment]
We were able to reproduce this issue once again on iPadOS 13.2 with a development app. Like before, toggling the wifi connection once again fixed the issue.

This time however I have captured the logs from Xcode > Window > "Devices and Simulators" and press the "Open Console".

```
default 17:24:12.757668-0800  apsd  _CFNetworkIsConnectedToInternet returning 1, flagsValid: 1, flags: 0x2
error 17:24:12.757893-0800  apsd  Task <8328A9BD-E011-4C05-B23C-FAB986B21615>.<1> HTTP load failed, 0/0 bytes (error code: -1003 [12:8])
default 17:24:12.757961-0800  apsd  Task <8328A9BD-E011-4C05-B23C-FAB986B21615>.<1> summary for task failure {transaction_duration_ms=23, response_status=0, cache_hit=0}
error 17:24:12.758038-0800  apsd  Task <8328A9BD-E011-4C05-B23C-FAB986B21615>.<1> finished with error [-1003] Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x139e1dc00 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 UserInfo={_kCFStreamErrorCodeKey=8, _kCFStreamErrorDomainKey=12}}, NSErrorFailingURLStringKey=<private>, NSErrorFailingURLKey=<private>, _kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, NSLocalizedDescription=A server with the specified hostname could not be found.}
default 17:24:12.758162-0800  apsd  <private> Calling configuration completion blocks, expiration date Wed Jan  8 17:39:12 2020 error: Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x139e1dc00 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 UserInfo={_kCFStreamErrorCodeKey=8, _kCFStreamErrorDomainKey=12}}, NSErrorFailingURLStringKey=<private>, NSErrorFailingURLKey=<private>, _kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, NSLocalizedDescription=A server with the specified hostname could not be found.}
default 17:24:12.758213-0800  apsd  removing all entries config 0x139db3b70
default 17:24:12.758341-0800  apsd  APSTCPStream <private> got config <private>  -- opened? YES
default 17:24:12.758805-0800  apsd  removing all entries config 0x139db3b70
default 17:24:12.759093-0800  apsd  Failed to load config bag: Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x139e1dc00 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 UserInfo={_kCFStreamErrorCodeKey=8, _kCFStreamErrorDomainKey=12}}, NSErrorFailingURLStringKey=<private>, NSErrorFailingURLKey=<private>, _kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, NSLocalizedDescription=A server with the specified hostname could not be found.}
default 17:24:12.759181-0800  apsd  Using DNS TXT record lookup to determine load balance count for development
default 17:24:12.759255-0800  apsd  <private> performing DNS lookup for <private>
```

In this instance it seems that some DNS look up continued to fail for an APNs connection, noted by the `apsd` process name at the start of each line.

For anyone else seeing the "APNS delegate never fired" error for you device it would be helpful you can capture the log as well. I recommend searching for "apsd  Failed to load config bag:" first to see if this matches something like the above. If this isn't present try just searching for "  apsd" or filter to the apsd process in the console.

**Comment by JinHoSo** [comment]
@jkasten2 I got the same error 

2020-01-22 19:50:49.585000+0900 TestApp[4199:1308229] VERBOSE: Calling OneSignal create/on_session
2020-01-22 19:50:49.586885+0900 TestApp[4199:1308229] VERBOSE: HTTP Request (OSRequestRegisterUser) with URL: https://onesignal.com/api/v1/players, with parameters: {
  "app_id" : "6d88c25b-8e37-4d35-b6ee-3278a5856379",
  "net_type" : 1,
  "test_type" : 1,
  "as_id" : "EA0025DC-FCBC-409C-9015-9603ADE09FF7",
  "ios_bundle" : "com.testApp.mobileapp",
  "device_type" : 0,
  "sdk_type" : "react",
  "sdk" : "021204",
  "language" : "ko-KR",
  "device_os" : "13.3",
  "game_version" : "1",
  "timezone" : 32400,
  "ad_id" : "8F7C215F-5202-4C92-9D40-87133701AF08",
  "notification_types" : -14,
  "device_model" : "iPhone8,4",
  "carrier" : "LG U+"
}
2020-01-22 19:50:50.159113+0900 TestApp[4199:1308207] VERBOSE: network response (OSRequestRegisterUser): {
    id = "50fa620f-499f-41cc-bb59-a51ef5afd71a";
    success = 1;
}
2020-01-22 19:50:50.159671+0900 TestApp[4199:1308207] Request <OSRequestRegisterUser: 0x2818e53b0> success result {
    id = "50fa620f-499f-41cc-bb59-a51ef5afd71a";
    success = 1;
}
2020-01-22 19:50:50.160299+0900 TestApp[4199:1307975] VERBOSE: on_session result: {
    push =     {
        id = "50fa620f-499f-41cc-bb59-a51ef5afd71a";
        success = 1;
    };
}
2020-01-22 19:50:50.166886+0900 TestApp[4199:1307975] VERBOSE: getNotificationTypes:mSubscriptionStatus: -1
2020-01-22 19:50:50.167085+0900 TestApp[4199:1307975] VERBOSE: getNotificationTypes:mSubscriptionStatus: -1
2020-01-22 19:51:04.677465+0900 TestApp[4199:1308381] Connection 20: received failure notification
2020-01-22 19:51:04.677550+0900 TestApp[4199:1308381] Connection 20: failed to connect 1:50, reason 0
2020-01-22 19:51:04.677592+0900 TestApp[4199:1308381] Connection 20: encountered error(1:50)
2020-01-22 19:51:04.679286+0900 TestApp[4199:1308381] Task <14920229-AACB-4EC2-9E3A-55F3F39FFC0C>.<1> HTTP load failed, 0/0 bytes (error code: -1009 [1:50])
2020-01-22 19:51:04.680278+0900 TestApp[4199:1307975] Task <14920229-AACB-4EC2-9E3A-55F3F39FFC0C>.<1> finished with error [-1009] Error Domain=NSURLErrorDomain Code=-1009 "The Internet connection appears to be offline." UserInfo={NSErrorFailingURLStringKey=https://play.googleapis.com/log/batch, _kCFStreamErrorDomainKey=1, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <14920229-AACB-4EC2-9E3A-55F3F39FFC0C>.<1>, NSURLErrorNetworkUnavailableReasonKey=0, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <14920229-AACB-4EC2-9E3A-55F3F39FFC0C>.<1>"
), NSLocalizedDescription=The Internet connection appears to be offline., NSErrorFailingURLKey=https://play.googleapis.com/log/batch, NSUnderlyingError=0x2818d89c0 {Error Domain=kCFErrorDomainCFNetwork Code=-1009 "(null)" UserInfo={NSURLErrorNetworkUnavailableReasonKey=0, _kCFStreamErrorCodeKey=50, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=50}
2020-01-22 19:54:17.721933+0900 TestApp[4199:1309020] [] tcp_input [C7.1:3] flags=[R] seq=866249271, ack=0, win=0 state=CLOSED rcv_nxt=866249271, snd_una=2172463612
2020-01-22 19:54:17.722527+0900 TestApp[4199:1309020] [] tcp_input [C7.1:3] flags=[R] seq=866249271, ack=0, win=0 state=CLOSED rcv_nxt=866249271, snd_una=2172463612
2020-01-22 19:54:17.722757+0900 TestApp[4199:1309020] [] tcp_input [C7.1:3] flags=[R] seq=866249271, ack=0, win=0 state=CLOSED rcv_nxt=866249271, snd_una=2172463612

**Comment by kajensen** [comment]
I have reproduced this as well. Toggling wi-fi fixed the issue.

**Comment by BastianBerlin** [comment]
> I have reproduced this as well. Toggling wi-fi fixed the issue.

If you mean turning wifi on and off , I tried that and it did not solve the problem. Plus - we can not expect that our users do that :)

(**running same build on an iPhone 5 with iOS 10.3.4 - no probs at all**; so I rather feel this issue is coming along with the new iOS version)

Is there anyone from OneSignal taking care of this? It seems since 12.2 there are massive problems with OneSignal / Ionic / iOS ...
Who has an idea? I'd love to hear!

**Comment by bharling** [comment]
Also seeing this ourselves over the last couple of days. Same error code for IOS, also seeing similar things for google play APNS.

Our apps are built on phonegap build as well

**Comment by BastianBerlin** [comment]
We need to have a call about this. Urgently.
A full week is lost again…

___






Helping you to get connected.

Sebastian Hesse

Office: 	+49 30 69202-499
Mobile: 	+49 1704769100
Skype: 	sebastian.bizzdar
sebastian.hesse@bizzdar.com
www.bizzdar.com

BIZZDAR GmbH
Winsstr. 61, 10405 Berlin

Company register: 
Amtsgericht Berlin (Charlottenburg)
HRB 202111 B

Managing Directors / Geschäftsführer:
Kumar Gaurav
Sebastian Hesse






Am 07.02.2020 um 17:57 schrieb Ben Harling <notifications@github.com>:

Also seeing this ourselves over the last couple of days. Same error code for IOS, also seeing similar things for google play APNS.

Our apps are built on phonegap build as well

**Comment by Neilfau** [comment]
Got the same issue, silent notifications do not trigger the delegate method.

**Comment by bharling** [comment]
We're seeing this error consistently now for all new builds via phonegap ( PGB and CLI locally ). We've tried plugin versions 2.4.5, 2.7.0 and 2.8.3 - all give the `Apns delegate never fired` error.

Is there any update to this issue at all ?

**Comment by bharling** [comment]
We seem to be getting somewhere using the 2.8.3 plugin version and making sure we use the legacy build system on xcode, it looks like something may have changed there recently. Unfortunately looks like the build flags needed are not supported on phonegap build so we're going to have to ditch that :(, it's a day of sadness as a good bit of our production pipeline depends on that .

**Comment by rgomezp** [comment]
It looks like many of you are seeing this specifically with Cordova & Cordova-based platforms. This leads me to believe this is not a native iOS issue, but rather specific to Cordova. Can we just confirm this please?

**Comment by Neilfau** [comment]
I’m using native development swift 5.0 & latest version of Xcode.

**Comment by kajensen** [comment]
I reproduced this on native iOS

**Comment by rgomezp** [comment]
Ok. This seems to be an Apple issue. My associate @jkasten2 will follow up here with more details

**Comment by lr-sanchez** [comment]
Same error here with onesignal-cordova-plugin 2.7.0 on device and emulator. Strange that month ago was working.

**Comment by jkasten2** [comment]
## Scope
Follow the below if you are seeing intermittent issues with some devices getting:
* "APNS delegate never fired" in the Xcodde log or on the OneSignal dashbaord user list
* If you called `[[UIApplication sharedApplication] registerForRemoteNotifications]` but `application:didFailToRegisterForRemoteNotificationsWithError` nor `application:didRegisterForRemoteNotificationsWithDeviceToken` is firing even after 60 seconds.
   - With our without OneSignal

**If you are consistently getting "APNS delegate never fired" on multiple devices then this points to a different issue. Most likely a setup issue with OneSignal, provisioning settings, or another library / SDK conflicting with OneSignal. *Please open a new ticket if this is the case.***

## Gathering more details 
Before creating a ticket with Apple I would like to get a few more details.

### 1. Logs with <private> filled in
Since the logs that were captured here did not have the "Messages Extensions Logging" profile installed on the iOS device we are not able to see some <private> fields such as the following.
```
default 17:24:12.759255-0800  apsd  <private> performing DNS lookup for <private>
```
These will be key to give the enough detail to Apple to debug the issue.

So before you see the issue please follow these steps to install the profiles
1. Go to https://developer.apple.com/bug-reporting/profiles-and-logs/ on your iOS device
2. Tap on "Profile" to the right of  "APNS (Apple Push Notification Service) for iOS".
  - Also recommend downloading "mDNSResponder for iOS", "CFNetwork for iOS" in case APNS doesn't provide all the details needed.
3. Follow it's instructions on downloading.
4. Lastly make sure go to Settings > General > Profiles and install it as Apple instructions.

When you reproduce the issue go to Xcode > Window > "Devices and Simulators" and press the "Open Console". Filter on `apsd`. If this is an issue you should see a log like the following.
https://github.com/OneSignal/OneSignal-iOS-SDK/issues/514#issuecomment-572373987

### 2. Reproduce with a production build
Since Apple uses it's sandbox server for development builds this might be just an issue with these servers. Please attach the logs if you can produce with a production build. By production it doesn't have to be a build on the App Store, it can be an ad-hoc, an .ipa that was built with production, or a test flight build.

### 3. (Optional) Reproduce issue with a custom DNS not resolving Apple's push server
If the device can't connect to Apple's APNS server it should be expected that `application:didFailToRegisterForRemoteNotificationsWithError` would fire with some kind of network error. If we can reproduce this consistently with a custom DNS then can suggest to Apple to add an error for this so App's have more detail on why the device could not register.

**Comment by rserro** [comment]
Maybe this can help someone: I was having this problem, tried serveral solutions, nothing. Then I found it was somehow related to having enabled Mac Catalyst on XCode. Turned it off and it worked.

**Comment by Apromodo** [comment]
Guys I found a solution if your user looks like in openSignal 

[APNS delegate never fired Error](https://quickup.in/i/080e.png)

you must add certificate in your iPhone
setting>general>profile and add push "apple Push Services"

[apple push servies](https://quickup.in/i/5b6b.png)

# For latest version of iPhone 



<img width="1609" alt="Ekran Resmi 2020-03-24 17 55 07" src="https://user-images.githubusercontent.com/51710685/77439854-a1ace100-6df8-11ea-93be-080ef855a574.png">


![IMG_6138](https://user-images.githubusercontent.com/51710685/77439601-4aa70c00-6df8-11ea-92f4-2b09d1955ad2.PNG)

**Comment by Lapinou42** [comment]
Hello guys, 

I got this issue today and found the solution after 3 hours -_-' 

I use Firebase/Crashlytics, RemoteConfig and Analytics. 
In my AppDelegate, I have to initialize Firebase FIRST then OneSignal.
After that, the issue gone away !

**Comment by rgomezp** [comment]
@Lapinou42 ,
Are you seeing this in production builds or just development?

**Comment by Lapinou42** [comment]
@rgomezp,
Indeed, I see this only in development.

**Comment by markosole** [comment]
Hi guys, I've just updated to latest MacOS, Xcode and iOS on my device. All works fine to me. Open up Xcode project, remove "Background push" what NS adds by itself and just add "Push notification". 
PS: reconfigure firebase plugin, clean platform etc. I did not try in production yet.

To be honest, I've seen errors in OneSignal Devices list, but after clearing and reconfiguring all, it works as it should.

**Comment by rgomezp** [comment]
Howdy @Lapinou42 @markosole ,
Christophe, this is likely a DNS issue only Apple can fix in that case.

Thanks for your suggestions Marko. Hopefully others can see the same success after trying your workaround.

**Comment by careboll72** [comment]
The solution from @markosole works for me... Only adding PUSH NOTIFICATION capability FROM XCODE...now I am not getting errors...

**Comment by markosole** [comment]
> The solution from @markosole works for me... Only adding PUSH NOTIFICATION capability FROM XCODE...now I am not getting errors...

I am not sure why NativeScript adds Background Notification, what for is that. That's probably main reason and nobody is checking that. I've tested it several times and work fine.

**Comment by rgomezp** [comment]
Howdy,
I believe it's legacy @markosole

**Comment by yigitbacakoglu** [comment]
Same issue here;

I'm trying to use OneSignal with latest ionic & capacitor build.

**Comment by rgomezp** [comment]
@yigitbacakoglu ,
Thanks for the update. This confirms our suspicions regarding it just being an issue with the sandbox servers. Apple will have to fix this.

**Comment by daisito** [comment]
The problem is solved by removing the wifi connection and testing with mobile data that turned out for me, despite the fact that I had already generated the certificate for the third time

**Comment by daisito** [comment]
![Captura de Pantalla 2020-05-12 a la(s) 11 06 10 p  m](https://user-images.githubusercontent.com/2993501/81766929-2e912400-94a5-11ea-8d16-bc5cbfd6c090.png)

**Comment by daisito** [comment]
Solved, it is a communication sandbox conflict with apple

![Captura de Pantalla 2020-05-12 a la(s) 11 06 43 p  m](https://user-images.githubusercontent.com/2993501/81766959-436db780-94a5-11ea-9da4-2659be84cc5e.png)

**Comment by beauchette** [comment]
I've read lots of issues about this `APNS Delegate Never Fired` problem. I'm currently trying to implement OneSignal (flutter) in my app, it's still in test. Basically, while developing it, I stumbled upon the issue rather quickly. and I discovered that basically, it works almost every time when I'm using wifi, and nearly never (not never) using 4G.

I read that it was an issue communicating with Apple development/sandbox APNs, so I made a TestFlight build, because it's supposed to use the production APNs, and I have the same exact problem. It doesn't work every time, and it works far better using wifi. When a user subscribes properly, I have no problem sending push notification.

I even tried uninstalling / installing my TestFlight build (while deleting the user on OneSignal's dashboard, each time) and sometimes it works, sometimes not. That's so confusing.

I'm using the lib in a really straightforward way, I use the feature requiring the user privacy consent, present the user a popup, and depending on his answer, I invoke `consentGranted(true)` and everything goes fine, except on certain iOS devices. Also noteworthy: I don't use any third party package that would make things fail.

**Comment by sumanc** [comment]
I also started running into this issue recently. One of my iPads consistently runs into this issue on Wi-Fi. There is no bandwidth issue. Lat time it happened, toggling Wi-Fi helped. 

It started again two days ago and I decided not to toggle Wi-Fi and leave it like that just in case OneSignal wants to debug or something. 

About APNS sandbox vs. production servers, I believe it is not the case. Because yesterday, Apple reported that they ran into this issue in my app when they were reviewing my latest build. I am assuming, when they are testing my app store build they hit production servers.

At this point, I am just waiting on OneSignal engineering team to respond.

**Comment by rgomezp** [comment]
I wonder if originally, this was indeed a sandbox server issue as people confirmed earlier. But it's possible that something changed which is leading to the same issue even on production environments. Would appreciate a recent stacktrace with OneSignal logging set to verbose.

**Comment by osmanc777** [comment]
How about you guys, this error is really a headache since a few days ago everything was working fine for me but suddenly I threw myself that error when trying to subscribe the user.

Today I managed to find a solution or at least make it work.

Steps to follow:

1. Have the Push Notifications added in Xcode
enable these options by the background push:
-Background Fetch
-Remote notifications

![xcodepush](https://user-images.githubusercontent.com/23218509/83215263-c6d50d00-a123-11ea-94a8-400d49a511bd.png)


2. Check if they have any updates to cocoa pods
```sh
pod repo update
```
![pods1](https://user-images.githubusercontent.com/23218509/83217643-7660ae00-a129-11ea-8386-0a1871b8c3ce.png)

```
sudo gem install cocoapods
```
![pods2](https://user-images.githubusercontent.com/23218509/83217698-94c6a980-a129-11ea-9aaa-9c32823e31c6.png)

then in the root folder of your project run again
```
pod repo update
```
![pods3](https://user-images.githubusercontent.com/23218509/83217765-b6279580-a129-11ea-9967-7431e6d4bf5e.png)

in my case as I am working with ionic, enable onesignal debugging options, building an object of each persimo that requests onesignal

```ts
      this.oneSignal.startInit('#####', ####');

      //important debuggin info
      this.oneSignal.setLogLevel({logLevel: 6, visualLevel: 0});

      this.oneSignal.inFocusDisplaying(this.oneSignal.OSInFocusDisplayOption.Notification);

      this.oneSignal.handleNotificationReceived().subscribe((respNoti) => {
        // do something when notification is received
      });

      this.oneSignal.handleNotificationOpened().subscribe((respNoti) => {
        // do something when a notification is opened
      });

      this.oneSignal.getIds().then(info => {
        this.userId = info.userId;
        this.userToken = info.pushToken;
        console.log('ID OneSignal: ',this.userId);
        console.log('Token push: ', this.userToken);
      });

      this.oneSignal.endInit();
```
//function get log status user (debug for safari)
```ts
  getStatusUser() {
    this.oneSignal.getPermissionSubscriptionState().then((resp:OSPermissionSubscriptionState) => {
      let Obj:any = {
        Promt:       resp.permissionStatus.hasPrompted,
        PermisosIOS: resp.permissionStatus.status,
        PermisosAnd: resp.permissionStatus.state,
        Suscrito:    resp.subscriptionStatus.subscribed,
        ConfigUser:  resp.subscriptionStatus.userSubscriptionSetting,
        UserID:      resp.subscriptionStatus.userId,
        UserToken:   resp.subscriptionStatus.pushToken
      }


      console.log('OneDebug: ',Obj);
      console.log('iOS: 0 = Not Determined, 1 = Denied, 2 = Authorized ', resp.permissionStatus.status);
      console.log('Android: 1 = Authorized, 2 = Denied ', resp.permissionStatus.state);
      console.log('FCM/APNs: ', resp.subscriptionStatus.pushToken);
    })
  }
```
![pushInfo](https://user-images.githubusercontent.com/23218509/83220030-2684e580-a12f-11ea-83d9-87927e6b5e77.png)



then proceed to generate project build again

to know if your push subscription was successful the userId and pushToken must contain a string of strings must not be null

basically the problem is fixed when updating cocoa pods.

**Comment by Lapinou42** [comment]
I got this issue on iOS Swift and on Xamarin.iOS. Not all the time, it’s random, very strange.

**Comment by beauchette** [comment]
As @sumanc @rgomezp state, it may have been a sandbox server issues at one point, this is not the case anymore.

@osmanc777 you don't even need `Background Fetch` with OneSignal. Also, don't believe you succeeded, it will randomly fail at one point. That's what's tricky about this. sometimes it's toggling wifi, sometimes updating your pods, etc. I tried all of this and more. And I had random failures each time.

I was just implementing push notifications on an app, I had this problem, circled around for days, and then I tried FCM instead of OneSignal, and since then, every device: sandbox or production, 4G, 3G, Wifi, etc always works fine

I prefer OneSignal's interface and functionality, I think it's easier to use (push notifications will be handled by your community manager when in production) we are really disappointed not to be able to use this clearly superior product. And I suppose that by going with FCM, it'll be a pain to change if we ever wanted to.

**Comment by pikzelz** [comment]
Issue is happening here now too. Updated to the latest version of OneSignal as I like to keep everything up to date and all of a sudden I see this "App Delegate Never Fired" popping up. While it did work before.

I have another client that's using OneSignal in their app that's already published and it runs flawlessly there. Looks like it's an issue with the newest version.

I though it might be related to Firebase as they use some method swizzling too, however after removing Firebase the issue is still there, any other third party framework is not included. It's very confusing as nothing weird shows up in the console.

**Comment by rgomezp** [comment]
Howdy,
One of our community members proposed [this workaround](https://github.com/OneSignal/OneSignal-iOS-SDK/issues/684) which apparently worked for them. Can you please give it a shot and upvote this comment if it also worked for you?

Since this seems to be an Apple problem, the best we can do currently is find a workaround.

**Comment by sumanc** [comment]
Sure, thanks for the tip. I will try that and keep an eye to make sure the issue went away.

**Comment by pikzelz** [comment]
That workaround unfortunately doesn’t work for me.

I noticed (in the simulator) that it sometimes reports iOS Simulator Not Supported (obviously) and then after second start (or background -> foreground). It happens to become App Delegate Never Fired, while it should stay Simulator not supported.

**Comment by daisito** [comment]
> That workaround unfortunately doesn’t work for me.
> 
> I noticed (in the simulator) that it sometimes reports iOS Simulator Not Supported (obviously) and then after second start (or background -> foreground). It happens to become App Delegate Never Fired, while it should stay Simulator not supported as it can’t even work with the simulator.

Works effectively on real device, test on real device with 3g connection the first time you will subscribe to onesignal

**Comment by daisito** [comment]
> > That workaround unfortunately doesn’t work for me.
> > I noticed (in the simulator) that it sometimes reports iOS Simulator Not Supported (obviously) and then after second start (or background -> foreground). It happens to become App Delegate Never Fired, while it should stay Simulator not supported as it can’t even work with the simulator.
> 
> Works effectively on real device, test on real device with 3g connection the first time you will subscribe to onesignal

4g or lte connection

**Comment by pikzelz** [comment]
> > That workaround unfortunately doesn’t work for me.
> > I noticed (in the simulator) that it sometimes reports iOS Simulator Not Supported (obviously) and then after second start (or background -> foreground). It happens to become App Delegate Never Fired, while it should stay Simulator not supported as it can’t even work with the simulator.
> 
> Works effectively on real device, test on real device with 3g connection the first time you will subscribe to onesignal

That doesn’t work either, as described earlier. There’s something weird happening within OneSignal, might be iOS 13.5 related.

**Comment by daisito** [comment]
i have used with IOS 13.5, and work for me

**Comment by daisito** [comment]
> > > That workaround unfortunately doesn’t work for me.
> > > I noticed (in the simulator) that it sometimes reports iOS Simulator Not Supported (obviously) and then after second start (or background -> foreground). It happens to become App Delegate Never Fired, while it should stay Simulator not supported as it can’t even work with the simulator.
> > 
> > 
> > Works effectively on real device, test on real device with 3g connection the first time you will subscribe to onesignal
> 
> That doesn’t work either, as described earlier. There’s something weird happening within OneSignal, might be iOS 13.5 related.

i have used with IOS 13.5, and work for me

**Comment by ArkaneKhan** [comment]
facing same issue in react native randomly getting Apns delegate never fired. on both wifi and 4g connections as mentioned by others toggling wifi sometime works. i have tested it with test flight build and i can confirm that its happening on test flight build as well. Even tried that background fetch workaround sadly it didn't help.
react native: 0.61.4
react-native-onesignal : 3.9.0
ios sdk through pod is OneSignal (2.14.3)

logs that i am gettting:

2020-07-22 11:07:05.372822+0500 gist[393:13797] VERBOSE: registerForPushNotifications Called:waitingForApnsResponse: 1
2020-07-22 11:07:05.376963+0500 gist[393:13572] VERBOSE: updateNotificationTypes called: 15
2020-07-22 11:07:05.377062+0500 gist[393:13572] VERBOSE: startedRegister: 1
2020-07-22 11:07:05.377096+0500 gist[393:13572] VERBOSE: Prompt For Push Notifications Success
2020-07-22 11:07:05.412364+0500 gist_social[393:13572] [] nehelper sent invalid result code [1] for Wi-Fi information request
2020-07-22 11:07:05.469 [info][tid:com.facebook.react.JavaScript] 'Device info: ', { pushToken: null,
  userId: '1444c5d5-0ede-4dc1-81e3-5ce618dd86hj' }

**Comment by JArmandoCl** [comment]
I have been facing this issue, but this https://github.com/OneSignal/OneSignal-iOS-SDK/issues/684 solve my problem now my ios user are subscribed to my app in onesignal dashboard, my test device was an iphone 8 ios 13.5

**Comment by chillok** [comment]
> I have been facing this issue, but this #684 solve my problem now my ios user are subscribed to my app in onesignal dashboard, my test device was an iphone 8 ios 13.5

Didn't solve it here.

**Comment by beauchette** [comment]
> I have been facing this issue, but this #684 solve my problem now my ios user are subscribed to my app in onesignal dashboard, my test device was an iphone 8 ios 13.5

I tried and it didn’t work for me either. It comes from firecloud messaging which makes you enable it, because it needs it. One signal does not

**Comment by chillok** [comment]
I really like OneSignal as a platform, and it's been a breeze to integrate our Web Service, but unfortunately it looks like we're just going to have to drop it immediately. This looks like it's been around with a long time and isn't going to go anywhere.

**Comment by chillok** [comment]
With regards [this comment](https://github.com/OneSignal/OneSignal-iOS-SDK/issues/514#issuecomment-572373987)

> We were able to reproduce this issue once again on iPadOS 13.2 with a development app. Like before, toggling the wifi connection once again fixed the issue.
> 
> This time however I have captured the logs from Xcode > Window > "Devices and Simulators" and press the "Open Console".
> 
> ```
> default 17:24:12.757668-0800  apsd  _CFNetworkIsConnectedToInternet returning 1, flagsValid: 1, flags: 0x2
> error 17:24:12.757893-0800  apsd  Task <8328A9BD-E011-4C05-B23C-FAB986B21615>.<1> HTTP load failed, 0/0 bytes (error code: -1003 [12:8])
> default 17:24:12.757961-0800  apsd  Task <8328A9BD-E011-4C05-B23C-FAB986B21615>.<1> summary for task failure {transaction_duration_ms=23, response_status=0, cache_hit=0}
> error 17:24:12.758038-0800  apsd  Task <8328A9BD-E011-4C05-B23C-FAB986B21615>.<1> finished with error [-1003] Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x139e1dc00 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 UserInfo={_kCFStreamErrorCodeKey=8, _kCFStreamErrorDomainKey=12}}, NSErrorFailingURLStringKey=<private>, NSErrorFailingURLKey=<private>, _kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, NSLocalizedDescription=A server with the specified hostname could not be found.}
> default 17:24:12.758162-0800  apsd  <private> Calling configuration completion blocks, expiration date Wed Jan  8 17:39:12 2020 error: Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x139e1dc00 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 UserInfo={_kCFStreamErrorCodeKey=8, _kCFStreamErrorDomainKey=12}}, NSErrorFailingURLStringKey=<private>, NSErrorFailingURLKey=<private>, _kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, NSLocalizedDescription=A server with the specified hostname could not be found.}
> default 17:24:12.758213-0800  apsd  removing all entries config 0x139db3b70
> default 17:24:12.758341-0800  apsd  APSTCPStream <private> got config <private>  -- opened? YES
> default 17:24:12.758805-0800  apsd  removing all entries config 0x139db3b70
> default 17:24:12.759093-0800  apsd  Failed to load config bag: Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x139e1dc00 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 UserInfo={_kCFStreamErrorCodeKey=8, _kCFStreamErrorDomainKey=12}}, NSErrorFailingURLStringKey=<private>, NSErrorFailingURLKey=<private>, _kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, NSLocalizedDescription=A server with the specified hostname could not be found.}
> default 17:24:12.759181-0800  apsd  Using DNS TXT record lookup to determine load balance count for development
> default 17:24:12.759255-0800  apsd  <private> performing DNS lookup for <private>
> ```
> 
> In this instance it seems that some DNS look up continued to fail for an APNs connection, noted by the `apsd` process name at the start of each line.
> 
> For anyone else seeing the "APNS delegate never fired" error for you device it would be helpful you can capture the log as well. I recommend searching for "apsd Failed to load config bag:" first to see if this matches something like the above. If this isn't present try just searching for " apsd" or filter to the apsd process in the console.
> 
> We were only able to find one other instance of a similar apsd error log noted below:
> https://stackoverflow.com/questions/59298620/apns-connection-failure

I'm seeing a different but related issue.

`
Failed to load config bag: Error Domain=NSURLErrorDomain Code=-1202 UserInfo={NSURLErrorFailingURLPeerTrustErrorKey=<private>, NSLocalizedRecoverySuggestion=Would you like to connect to the server anyway?, _kCFStreamErrorDomainKey=3, _kCFStreamErrorCodeKey=-9807, NSErrorPeerCertificateChainKey=<private>, NSUnderlyingError=0x111f44700 {Error Domain=kCFErrorDomainCFNetwork Code=-1202 UserInfo={_kCFStreamPropertySSLClientCertificateState=0, kCFStreamPropertySSLPeerTrust=<private>, _kCFNetworkCFStreamSSLErrorOriginalValue=-9807, _kCFStreamErrorDomainKey=3, _kCFStreamErrorCodeKey=-9807, kCFStreamPropertySSLPeerCertificates=<private>}}, NSLocalizedDescription=<private>, NSErrorFailingURLKey=<private>, NSErrorFailingURLStringKey=<private>, NSErrorClientCertificateStateKey=0}
`

Which looks like some certificate issue 🤔 Maybe there are two issues here rather than one?

**Comment by canado** [comment]
Hello,

If never I have a real iOS device but no access to LTE, ( only wifi ) and that I have the Apns Delegate Never Fired error.
What are my options to QA my app and check that everything is working before publishing the app on the app store?

**Comment by canado** [comment]
Any evolution, 100% proof solutions?
Thanks

**Comment by wangxingxing123654** [comment]
another library / SDK conflicting with OneSignal.
because our app has intergrated with the livechat[https://docs.kommunicate.io/docs/ios-installation.html] which include the the push code
`
import UIKit
import Flutter
import UserNotifications
UIApplicationMain
objc class AppDelegate: FlutterAppDelegate {

override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
) -> Bool {
    GeneratedPluginRegistrant.register(with: self)
    UNUserNotificationCenter.current().delegate = self
    KommunicateWrapper.shared.application(application, didFinishLaunchingWithOptions: launchOptions)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
}

override func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
    KommunicateWrapper.shared.userNotificationCenter(center, willPresent: notification, withCompletionHandler: { options in
        completionHandler(options)
    })
}

override func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
    KommunicateWrapper.shared.userNotificationCenter(center, didReceive: response, withCompletionHandler: {
        completionHandler()
    })
}

override func applicationWillTerminate(_ application: UIApplication) {
    KommunicateWrapper.shared.applicationWillTerminate(application: application)
}

override func applicationWillEnterForeground(_ application: UIApplication) {
    print("APP_ENTER_IN_FOREGROUND")
    UIApplication.shared.applicationIconBadgeNumber = 0
}

override func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
    print("DEVICE_TOKEN_DATA :: \(deviceToken.description)") // (SWIFT = 3) : TOKEN PARSING
    KommunicateWrapper.shared.application(application, didRegisterForRemoteNotificationsWithDeviceToken: deviceToken)
}
}

`

i have checked the https://documentation.onesignal.com/docs/troubleshooting-ios#section-apns-delegate-never-fired ,it

it seems that reason 3 [
3 - Other Dependencies, especially other Push Provider SDKs, can interfere with this. It is best to remove the other Push SDK from the app.] caused the problem,
i removed the km chat about notification code ,the onesignal doesnot show apns delegate never fired, onesignal worked again.
what should i do to avoid this??

**Comment by SigniOS** [comment]
Zsign technical support:
Apple signature tool supporting ios15 and ios16
Support the removal of dylib dynamic library
Completely free, permanently free!
https://urlzs.com/44zCx

**Comment by emawby** [comment]
Closing due to inactivity, but please reopen if anyone is still experiencing this issue
